jokusoramame/plugins/gambling.py

```diff
@@ -0,0 +1,188 @@
+# TODO
+# - item system
+#   - store
+#   - use
+#   - inventory
+
+import random
+
+import numpy as np
+import tabulate
+from curious import Embed, Member
+from curious.commands import Context, Plugin, command
+from curious.commands.decorators import ratelimit
+from curious.ext.paginator import ReactionsPaginator
+
+from jokusoramame.db.tables import UserBalance
+from jokusoramame.utils import chunked
+
+BAD_RESPONSES = [
+    '\N{FIRE} Your bank account went up in flames and you lost **{0} :̶.̶|̶:̶;̶**.',
+    '\N{SHOPPING TROLLEY} You spend **{0} :̶.̶|̶:̶;̶** at the supermarket.',
+    '\N{GHOST} A ghost pops up out of nowhere and scares you. You drop **{0} :̶.̶|̶:̶;̶**.',
+    '\N{HAMMER AND PICK} _All that is solid melts into air._ (You lost **{0} :̶.̶|̶:̶;̶**.)',
+    '\N{DRAGON} Ryuu ga waga teki wo kurau! (You lost **{0} :̶.̶|̶:̶;̶**.)'
+]
+
+GOOD_RESPONSES = [
+    '\N{FIRST PLACE MEDAL} You win first place in the Money Making Race™ and gain **{0} :̶.̶|̶:̶;̶**.',
+    '\N{SLOT MACHINE} You have a gambling addiction and win **{0} :̶.̶|̶:̶;̶**.',
+    '\N{GEM STONE} You find a diamond and sell it for **{0} :̶.̶|̶:̶;̶**.'
+]
+
+
+async def ensure_balance(ctx: Context, member: Member = None):
+    """
+    Return a balance associated to the member and corresponding guild. If one does not exist it is created.
+
+    :param ctx: Context object
+    :param member: Member to ensure balance for, defaults to author.
```

Comment:
This shouldn't be part of this function, if it's a utility function. Really, the bot and member should be passed in, and not the ctx (for easier 1.0+ porting when it goes away).